packages/babel-parser/src/parser/statement.ts

```diff
@@ -3087,27 +3069,81 @@ export default abstract class StatementParser extends ExpressionParser {
       }
       node.value = this.parseStringLiteral(this.state.value);
       this.finishNode(node, "ImportAttribute");
-      attrs.push(node);
+      attrs.push(node as N.ImportAttribute);
     } while (this.eat(tt.comma));
 
     return attrs;
   }
 
-  maybeParseImportAssertions() {
-    // [no LineTerminator here] AssertClause
-    if (this.isContextual(tt._assert) && !this.hasPrecedingLineBreak()) {
-      this.expectPlugin("importAssertions");
+  expectImportAttributesPlugin() {
+    if (!this.hasPlugin("importAssertions")) {
+      this.expectPlugin("importAttributes");
+    }
+  }
+
+  maybeParseImportAttributes(
+    node: Undone<N.ImportDeclaration | N.ExportNamedDeclaration>,
+  ) {
+    let attributes: N.ImportAttribute[];
+    let useWith = false;
+
+    if (this.match(tt._with)) {
```

Comment:
```suggestion
    // https://tc39.es/proposal-import-attributes/#prod-WithClause
    if (this.match(tt._with)) {
```
Nit. Attaching the spec link here for reference.